event_rest_api/services/event.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2021 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _
+from odoo.exceptions import ValidationError
+
+from odoo.addons.base_rest import restapi
+from odoo.addons.base_rest_pydantic.restapi import PydanticModel, PydanticModelList
+from odoo.addons.component.core import Component
+
+from ..pydantic_models.event_info import EventInfo
+from ..pydantic_models.event_registration_info import EventRegistrationInfo
+from ..pydantic_models.event_registration_param import EventRegistrationParamList
+from ..pydantic_models.event_search_param import EventSearchParam
+
+
+class EventService(Component):
+    _inherit = "base.event.rest.service"
+    _name = "event.rest.service"
+    _usage = "event"
+    _expose_model = "event.event"
+    _description = __doc__
+
+    @restapi.method(
+        routes=[(["/<int:id>"], "GET")],
+        output_param=PydanticModel(EventInfo),
+        auth="public",
+    )
+    def get(self, _id):
+        event = self._get(_id)
+        return EventInfo.from_orm(event)
+
+    @restapi.method(
+        routes=[(["/", "/search"], "GET")],
+        input_param=PydanticModel(EventSearchParam),
+        output_param=PydanticModelList(EventInfo),
+        auth="public",
+    )
+    def search(self, event_search_param):
+        domain = []
+        if event_search_param.name:
+            domain.append(("name", "like", event_search_param.name))
+        if event_search_param.id:
+            domain.append(("id", "=", event_search_param.id))
+        res = []
+        for e in self.env["event.event"].sudo().search(domain):
+            res.append(EventInfo.from_orm(e))
+        return res
+
+    def _prepare_event_registration_values(self, event, event_registration_param):
+        return {
+            "event_id": event.id,
+            "partner_id": self.env.context["authenticated_partner_id"],
+            "firstname": event_registration_param.firstname,
+            "lastname": event_registration_param.lastname,
+            "email": event_registration_param.email,
+            "phone": event_registration_param.phone,
+            "event_ticket_id": event_registration_param.event_ticket_id,
+        }
+
+    @restapi.method(
+        routes=[(["/<int:id>/registration"], "POST")],
+        input_param=PydanticModel(EventRegistrationParamList),
```

Comment:
Why not `PydanticModelList(EventRegistrationParam)`?

Reply:
Because lists are not supported as parameters.
I can't figure out how to solve this. Or is PydanticModelList the way to go ?

```
Traceback (most recent call last):
  File "/home/quentin/.virtualenvs/odoo-ecolo-14/lib/python3.6/site-packages/odoo/addons/base/models/ir_http.py", line 237, in _dispatch
    result = request.dispatch()
  File "/home/quentin/.virtualenvs/odoo-ecolo-14/lib/python3.6/site-packages/odoo/http.py", line 807, in dispatch
    r = self._call_function(**self.params)
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/quentin/Documents/oca/rest-framework/base_rest/http.py", line 182, in _handle_exception
    return super(HttpRestRequest, self)._handle_exception(exception)
  File "/home/quentin/.virtualenvs/odoo-ecolo-14/lib/python3.6/site-packages/odoo/http.py", line 745, in _handle_exception
    return super(HttpRequest, self)._handle_exception(exception)
  File "/home/quentin/.virtualenvs/odoo-ecolo-14/lib/python3.6/site-packages/odoo/http.py", line 316, in _handle_exception
    raise exception.with_traceback(None) from new_cause
TypeError: _call_function() argument after ** must be a mapping, not list
```